### ISSUE TITLE
PRIORITY 1 - alert test emails seem to have incorrect logic

### ISSUE THREAD

**Issue by iddmaxwell** [iddmaxwell/optimisebills#83]
All of the emails report the same - that the tariff has ended and the user is on the standard rate. 
This is incorrect. 
For example Alistair Groom who has recently switched to Iresa. 
It says he can save £245 by switching. 
If you click on the link in the email it reports correctly that there is no saving available. 
**(A separate issue is that this comparison page says he can switch to a fixed rate for an extra £48. This appears to be the Tonik tariff although it is not the cheapest alternative fixed tariff. The Iresa tariff he is on is already fixed so this is misleading. If the client is already on a fixed rate it shouldn't suggest moving to a different one unless it's cheaper).** 
There seems to be something wrong with the comparison page not identifying fixed tariffs correctly. 

Not sure where the email generating logic is getting the tariff end date. The api for the comparison page seems to calculate correctly so why not use that?

**Comment by iddmaxwell** [comment]
client email ended 1
Dear {Salutation}
Thank you very much for being a client of OptimiseBills.

According to my records your previous tariff has ended and you are no longer on the best rate.
You are with {Supplier} but now on the {Standard} tariff.

This costs about {monthlycost} per month, using {gaskwh} kWh of gas and {eleckwh} kWh of electricity per year 
{or} {gaskwh} kWh of gas per year 
{or} {eleckwh} kWh of electricity per year.

You could save up to {maxsaving} over 12 months by switching now.
Please click below or call 0345 130 3858 or reply to this email.

{BUTTON: see how much I could save}

Kind regards
{adminname}

**Comment by iddmaxwell** [comment]
client email ending 1
Dear {Salutation}
Thank you very much for being a client of OptimiseBills.

According to my records your tariff will be ending and you will no longer be on the best rate.
You are with {Supplier} on the {Tariff} tariff.  

You could save up to {maxsaving} over 12 months by switching now.
Please click below or call 0345 130 3858.

{BUTTON: see how much I could save}

If your details have changed, please let me know by replying to this email or calling 0345 130 3858. 

Kind regards
{adminname}

**Comment by iddmaxwell** [comment]
client email "Better"

Dear {Salutation}
Thank you very much for being a client of OptimiseBills.

According to my records you are with {Supplier} on the {Tariff} tariff.

You could save up to {maxsaving} over 12 months by switching now.  The terms of your current tariff are: "{terms relating to cancellation fees - text including and following the word 'cancellation'}

Please click below or call 0345 130 3858.

{BUTTON: see how much I could save}

If your details have changed, please let me know by replying to this email or calling 0345 130 3858.

Kind regards
{adminname}

**Comment by iddmaxwell** [comment]
Hi Will 
I’ve updated issue 87 with the text for the emails on tariffs ending, using curly brackets.  Hope it makes sense.

On the ‘better’ template, which is aimed at clients mid-term, cancellation fees could apply.  I looked at the API and it seems that cancellation fee terms appear in the quotes feed but not in the tariff feed.

I assume that the quotes feed provides the cancellation fees for the client’s current tariff.  On the UK Power web site, it is shown as we discussed on our call last week.

If not, we would have to store the cancellation fee terms on the client comparison record when the initial comparison is set up.  (However I don’t know then how we’d handle new clients coming to us mid-term to determine what their cancellation fees are.  Probably manually)

Assuming we can access cancellation fees via the API quotes feed – or from our client energy record – then we can show that text in the email so the client is aware that cancellation fees could apply.  You are also going to grab the tariff end date from the tariff feed.  So we know that cancellation fees could apply up until x date.  

If the API works differently and you can’t produce the BETTER email you’ll have to let me know so we can find another solution.

**Comment by Will-Newton** [comment]
Exit fees for the current tariff aren't on the api so I don't think we can do that. WHat does the top saving need to be in order to send the Better email? Are we still having the ending 2 email? I think it's basically there now in terms of accuracy, I just need to check what's causing some to fail and check why

**Comment by iddmaxwell** [comment]
client email "Best" - i.e. if {current_cost} <= {min_cost} and if {expiry_date} is over 42 days away
Dear {Salutation}
Thank you very much for being a client of OptimiseBills.
You are already on the best deal! 
According to my records you are with {Supplier} on the {Tariff} tariff which runs until {expiry_date}
You are currently spending about {current_cost}/12 per month and the cheapest alternative is would be {min_cost}/12 instead.

You can see the latest deals here: 
{BUTTON: see how much I could save}
If your details have changed, please let me know by replying to this email or calling 0345 130 3858.
Kind regards
{adminname}

**Comment by iddmaxwell** [comment]
Can we have the kwh values formatted with commas as 25,062 kWh rather than 25062?

**Comment by iddmaxwell** [comment]
Is it possible to include a line in the mid term email IF there is a cancellation fee (which is on the tariff record): 
Please bear in mind you could be charged a cancellation fee of up to £x if you switch before your tariff ends.

**Comment by Will-Newton** [comment]
Are these all minor issues now? If we can get the logic for selecting clients we can probably get this up and running and then look at the minor issues. What logic do we need for selecting clients? 

I was thinking somehting like:
Send 10 per day...
Find the 10 with the highest priority (order: ended/ending/better/best) that haven't been notified at that haven't previously been notified at that level of priority.

And then maybe something for ended clients that have previously been notified but still haven't respodned?

**Comment by iddmaxwell** [comment]
For tariffs ending
Send if end date is within 49!days and we haven't sent an email to the same client les than 8 days ago 

Better tariffs. Send monthly


Ended tariffs send every 14 days. 

Priority is ending then ended then better then best. 

Best emails 
Send Quarterly

**Comment by Will-Newton** [comment]
Is this including people who haven't signed up?

**Comment by iddmaxwell** [comment]
Yes can include people not yet signed up. But not so often. Maybe weekly after registration for a month then quarterly. 

Cam you

**Comment by iddmaxwell** [comment]
We need different wording for non clients. First line says thank you for being a client. 

Also can you switch off the automatic repetition of the same test emails unless you've changed something? In which case I want to see the new email to test it.

**Comment by Will-Newton** [comment]
Including non-clients will bring up users from years ago. So they'll go in as tariff ended. Do we need to re-think this as a lot fo the data is probably completely wrong?

Also, how do we determine non-clients now? Did you say it was in the status field?

**Comment by Will-Newton** [comment]
We're approaching this in the wrong way. We needed to just come up with some logic to grab a handful of users each day*. We'd then pass them through the api, if they saved £?00 amount of money then send them an email saying switch now. 

Then look at which bits are well received (if any) before considering a more comprehensive solution, or just a second email template or leaving it where it is to focus on something else. 

By tackling 4 scenarios from start to finish at kick off, it's already looking like another maintenance overhead as well as requiring loads of testing. It's also added requirements like putting everyone through the api initially just to get end dates and that in turn has thrown up such issues as the one where changes made on the old website to end dates need to be picked up.

I've done quite a lot of this now, so will carry on but please can we bear this in mind for new developments?

*I'm still not sure how we need to do this for the initial run as there won't be any previous sent dates? Just split all signed up clients by 365 so one or two go out per day?
